cluster-state-service/internal/features/e2etasksteps/common_steps.go

```diff
@@ -34,48 +38,109 @@ var (
 	cssTaskList   = []models.Task{}
 	exceptionList = []Exception{}
 
-	taskDefnARN = ""
+	taskDefnARN         = ""
+	roleName            = "E2ETestRole"
+	instanceProfileName = "E2ETestInstance"
+	policyARN           = "arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceforEC2Role"
 )
 
+const errorCode = 1
+
 func init() {
 
 	cssWrapper := wrappers.NewCSSWrapper()
 	ecsWrapper := wrappers.NewECSWrapper()
+	ec2Wrapper := wrappers.NewEC2Wrapper()
+	iamWrapper := wrappers.NewIAMWrapper()
 
 	BeforeAll(func() {
-		var err error
+		clusterName := wrappers.GetClusterName()
+
+		err := ecsWrapper.CreateCluster(&clusterName)
+		if err != nil {
+			fmt.Println(err.Error())
+			os.Exit(1)
+		}
+
+		err = terminateAllContainerInstances(ec2Wrapper, ecsWrapper, clusterName)
+		if err != nil {
+			fmt.Println(err.Error())
+			os.Exit(1)
+		}
+
 		taskDefnARN, err = ecsWrapper.RegisterSleep360TaskDefinition()
 		if err != nil {
-			T.Errorf(err.Error())
+			fmt.Println(err.Error())
+			os.Exit(1)
+		}
+
+		err = createInstanceProfile(iamWrapper)
+		if err != nil {
+			fmt.Println(err.Error())
+			os.Exit(1)
+		}
+
+		amiID, err := wrappers.GetLatestECSOptimizedAMIID()
+		if err != nil {
+			fmt.Println(err.Error())
+			os.Exit(1)
+		}
+
+		keyPair := wrappers.GetKeyPairName()
+		numInstances := int64(1)
+		_, err = ec2Wrapper.RunInstance(&numInstances, &clusterName, &instanceProfileName, &keyPair, &amiID)
+		if err != nil {
+			fmt.Println(err.Error())
+			os.Exit(1)
+		}
+
+		err = wrappers.ValidateECSContainerInstance(ecsWrapper, clusterName)
+		if err != nil {
+			fmt.Println(err.Error())
+			os.Exit(1)
 		}
 	})
 
 	Before("@task|@stream-instances", func() {
-		clusterName, err := wrappers.GetClusterName()
+		clusterName := wrappers.GetClusterName()
+		err := stopAllTasks(ecsWrapper, clusterName)
```

Comment:
`T` below will not be initialized until after the first test. You probably want to check if `T` is not nil below so that it doesn't panic.